io/io/src/TBufferMerger.cxx

```diff
@@ -54,10 +54,11 @@ TBufferMerger::~TBufferMerger()
 
 std::shared_ptr<TBufferMergerFile> TBufferMerger::GetFile()
 {
-   R__LOCKGUARD(gROOTMutex);
    std::shared_ptr<TBufferMergerFile> f(new TBufferMergerFile(*this));
-   gROOT->GetListOfFiles()->Remove(f.get());
-   fAttachedFiles.push_back(f);
+   {
+      R__LOCKGUARD(gROOTMutex); // This mutex can be replaced by a local one?
```

Comment:
Yes, if the file is not registered by ROOT we could just use the `fQueueMutex` here, or add a new mutex just for this.